executor/src/witgen/jit/block_machine_processor.rs

```diff
@@ -816,4 +816,68 @@ S::Z[0] = params[1];
 params[2] = S::carry[0];"
         );
     }
+
+    #[test]
+    fn bit_decomp_negative_concrete() {
+        let input = "
+        namespace Main(256);
+            col witness a, b, c;
+            [a, b, c] is [S.Y, S.Z,  S.carry];
+        namespace S(256);
+            let BYTE: col = |i| i & 0xff;
+            let X;
+            let Y;
+            let Z;
+            Y = 19;
+            Z = 16;
+            let carry;
+            carry * (1 - carry) = 0;
+            [ X ] in [ BYTE ];
+            [ Y ] in [ BYTE ];
+            [ Z ] in [ BYTE ];
+            X + Y = Z + 256 * carry;
+        ";
+        let code = format_code(&generate_for_block_machine(input, "S", 2, 1).unwrap().code);
+        assert_eq!(
+            code,
+            "\
+S::Y[0] = params[0];
+S::Z[0] = params[1];
+S::X[0] = 253;
+S::carry[0] = 1;
+params[2] = 1;"
```

Comment:
noob: what is params here?

Reply:
params is the array of bus receive parameters / lookup parameters. Special variables are needed because you could have complex expressions: `[a, b] in [x + 2, 2 * y]`. Assuming the first one is an input and the second one is an output, then we have:
`param[0]` is known -> `x = param[0] - 2`. At some point we compute `y` and then we can solve `param[1] = y / 2`, and that is what is returned from the submachine call.

Reply:
Great thanks, that makes sense